### ISSUE TITLE
Allow pgModeler to rename tables and fields to prevent loss of data

### ISSUE THREAD

**Issue by dpomerantz-jci** [pgmodeler/pgmodeler#1795]
**Feature description**
Allow pgModeler to rename tables and fields to prevent loss of data

**Additional info**
This appears to be the same issue as Field Rename #592 

Currently, when a table or column is renamed, the name is changed in the DBM file, and the next time the schema is compared against a database, it results in the object with the old name being dropped and a new object with the new name being created.  This makes sense as it's what pgModeler sees.  The way around this is for the database development team to deploy a script before the DBM sync which manually renames objects.  This works, but is an extra step of manual database changes, which pgModeler is intended to reduce.

I would like a feature added to pgModeler where a rename is persisted in the DBM file, and if the compare notices a drop and recreate, it issues a rename instead.  Essentially, whenever a Rename occurs in pgModeler, the previous name is stored in the XML document for that element.  Something like this:

<table name="customer" previous-name="customers">
	<column name="customer_id" previous-name="customers_id">
</table>

Then, when the sync happens, if it notices a drop and create for customers/customer or customers_id/customer_id, it issues an alter table rename isntead.

Optionally, instead of having an attribute with a single previous-name value, there could be an element <previous-names> with multiple values.  Ths is an implementation detail.

**Comment by rkhaotix** [comment]
Great suggestion, @dpomerantz-jci 

I'll think of a way to implement that and expand the idea for other object kinds and not restrict it to columns.

**Comment by dpomerantz-jci** [comment]
Thank you.  It's completely up to you, but I'm not sure if it's worth doing this for other object types.  Tables and fields contain data, so dropping them loses data.  Maybe add schemas to that?  If a view or a stored procedure has to be completely dropped and recreated on a rename, or even an index or constraint, it's not the end of the world.  Just MHO.  Thanks again for putting this on the backlog!

**Comment by rkhaotix** [comment]
> Thank you. It's completely up to you, but I'm not sure if it's worth doing this for other object types. Tables and fields contain data, so dropping them loses data. Maybe add schemas to that? If a view or a stored procedure has to be completely dropped and recreated on a rename, or even an index or constraint, it's not the end of the world. Just MHO. Thanks again for putting this on the backlog!

Yeah, I think you have a point here... I'll focus first on columns! :)
